### PR TITLE
Use torchaudio for sox effects

### LINKED ISSUE #14: Migrate to torchaudio's sox implementation
torchaudio 0.7 -- coming out tomorrow :) -- now has a `apply_effects_tensor` transform that applies any sox effects directly to pytorch tensors. In addition, the C++ bindings are now using native pytorch C++ extension, thus making the operations jitable. Please see [here](https://github.com/pytorch/audio/blob/master/torchaudio/sox_effects/sox_effects.py#L61) for an example of how this is done.

@eugene-kharitonov -- as we had discussed offline, we can help with the migration.

### PR DESCRIPTION
**Pull request by vincentqb**
As mentioned in #14, here's a pull request migrating WavAugment to torchaudio 0.7 :) This replaces the C++ backend with torchaudio. 

- Tested successfully on linux and macos. sox is not supported on windows.
- The list of KNOWN_EFFECTS now includes `{'fir'}` and removes `{'input', 'noisered', 'ladspa', 'output'}` in this pull request.
- torchaudio uses torchbinds (instead of pybind11), and so `torchaudio.sox_effects.apply_effects_tensor` is jitable.


```python
❯ python -m pytest -v --doctest-modules
=============================================================================================================== test session starts ================================================================================================================
platform linux -- Python 3.8.6, pytest-6.1.1, py-1.9.0, pluggy-0.13.1 -- /private/home/vincentqb/anaconda3/envs/wavaugment/bin/python
cachedir: .pytest_cache
rootdir: /private/home/vincentqb/WavAugment
collected 14 items                                                                                                                                                                                                                                 

augment/effects.py::augment.effects.EffectChain PASSED                                                                                                                                                                                       [  7%]
augment/effects.py::augment.effects.EffectChain.additive_noise PASSED                                                                                                                                                                        [ 14%]
augment/effects.py::augment.effects.EffectChain.clip PASSED                                                                                                                                                                                  [ 21%]
augment/effects.py::augment.effects.EffectChain.time_dropout PASSED                                                                                                                                                                          [ 28%]
augment/effects.py::augment.effects.SoxEffect.instantiate PASSED                                                                                                                                                                             [ 35%]
tests/augment_test.py::test_empty_chain PASSED                                                                                                                                                                                               [ 42%]
tests/augment_test.py::test_non_empty_chain PASSED                                                                                                                                                                                           [ 50%]
tests/augment_test.py::test_agains_cl PASSED                                                                                                                                                                                                 [ 57%]
tests/augment_test.py::test_stochastic_pitch PASSED                                                                                                                                                                                          [ 64%]
tests/augment_test.py::test_additive_noise PASSED                                                                                                                                                                                            [ 71%]
tests/augment_test.py::test_number_effects PASSED                                                                                                                                                                                            [ 78%]
tests/compare_to_sox_test.py::test_pitch PASSED                                                                                                                                                                                              [ 85%]
tests/compare_to_sox_test.py::test_reverb PASSED                                                                                                                                                                                             [ 92%]
tests/compare_to_sox_test.py::test_bandreject PASSED                                                                                                                                                                                         [100%]

================================================================================================================= warnings summary =================================================================================================================
../anaconda3/envs/wavaugment/lib/python3.8/site-packages/torchaudio/backend/utils.py:53
  /private/home/vincentqb/anaconda3/envs/wavaugment/lib/python3.8/site-packages/torchaudio/backend/utils.py:53: UserWarning: "sox" backend is being deprecated. The default backend will be changed to "sox_io" backend in 0.8.0 and "sox" backend will be removed in 0.9.0. Please migrate to "sox_io" backend. Please refer to https://github.com/pytorch/audio/issues/903 for the detail.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
========================================================================================================== 14 passed, 1 warning in 1.34s ===========================================================================================================
```

Thoughts?

Closes #14

cc @eugene-kharitonov @astaff

### CODE REVIEW

File: augment/effects.py [hunk 1]
```diff
@@ -7,16 +7,30 @@
 
 import torch
 import numpy as np
-import augment_cpp as _augment
+import torchaudio
+from torchaudio.sox_effects.sox_effects import effect_names as get_effect_names
 
 def shutdown_sox() -> None:
-    _augment.shutdown_sox()
+    pass
```
Comment (vincentqb) [inline augment/effects.py:14]: torchaudio now takes care of automatically initializing and shutting down sox. I kept it here to minimize the changes to the tests.

**Comment by vincentqb** [comment]
BTW the warning simply means that `torchaudio.set_audio_backend("sox_io")` is the default on master and will be the default in the next release.

**Comment by eugene-kharitonov** [comment]
Cool, thanks a lot @vincentqb ! I'll check this out in the next few days  :)

**Review by eugene-kharitonov — APPROVED** [review]

**Comment by vincentqb** [comment]
yay for the approval :)

**Pull request merged**